core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -0,0 +1,237 @@
+             pubSubDao: GooglePubSubDAO,
+             avroUpsertPubSubProject: GoogleProject,
+             pubSubTopicName: String,
+             pubSubSubscriptionName: String,
+             avroUpsertBucketName: String,
+             batchSize: Int,
+             workerCount: Int)(implicit executionContext: ExecutionContext, cs: ContextShift[IO]): Props =
+    Props(
+      new AvroUpsertMonitorSupervisor(
+        pollInterval,
+        pollIntervalJitter,
+        workspaceService,
+        googleServicesDAO,
+        samDAO,
+        googleStorage,
+        pubSubDao,
+        avroUpsertPubSubProject,
+        pubSubTopicName,
+        pubSubSubscriptionName,
+        avroUpsertBucketName,
+        batchSize,
+        workerCount))
+}
+
+class AvroUpsertMonitorSupervisor(
+                                       val pollInterval: FiniteDuration,
+                                       pollIntervalJitter: FiniteDuration,
+                                       workspaceService: UserInfo => WorkspaceService,
+                                       googleServicesDAO: GoogleServicesDAO,
+                                       samDAO: SamDAO,
+                                       googleStorage: GoogleStorageService[IO],
+                                       pubSubDao: GooglePubSubDAO,
+                                       avroUpsertPubSubProject: GoogleProject,
+                                       pubSubTopicName: String,
+                                       pubSubSubscriptionName: String,
+                                       avroUpsertBucketName: String,
+                                       batchSize: Int,
+                                       workerCount: Int)(implicit cs: ContextShift[IO])
+  extends Actor
+    with LazyLogging {
+  import AvroUpsertMonitorSupervisor._
+  import context._
+
+  self ! Init
+
+  override def receive = {
+    case Init => init pipeTo self
+    case Start => for (i <- 1 to workerCount) startOne()
+    case Status.Failure(t) => logger.error("error initializing avro upsert monitor", t)
+  }
+
+  def topicToFullPath(topicName: String) = s"projects/${avroUpsertPubSubProject.value}/topics/${pubSubTopicName}"
+
+  def init =
+    for {
+      _ <- pubSubDao.createSubscription(pubSubTopicName, pubSubSubscriptionName, Some(600)) //TODO: read from config
+    } yield Start
+
+  def startOne(): Unit = {
+    logger.info("starting AvroUpsertMonitorActor")
+    actorOf(AvroUpsertMonitor.props(pollInterval, pollIntervalJitter, workspaceService, googleServicesDAO, samDAO, googleStorage, pubSubDao, pubSubSubscriptionName, avroUpsertBucketName, batchSize))
+  }
+
+  override val supervisorStrategy =
+    OneForOneStrategy() {
+      case e => {
+        logger.error("unexpected error in avro upsert monitor", e)
+        // start one to replace the error, stop the errored child so that we also drop its mailbox (i.e. restart not good enough)
+        startOne()
+        Stop
```

Comment:
Does this stop the monitor, monitor supervisor, or both? Since `startOne` is called, this feels like it's in the context of the monitor supervisor, which gives me the impression that this `Stop` applies to the monitor supervisor as well... though I haven't spent any time learning about supervisor strategies so maybe I'm making poor assumptions about the contract of this function.

Reply:
I'm not well-versed in Akka either but I believe it just stops the particular child. From the `OneForOneStrategy` docs:

``` 
* Applies the fault handling `Directive` (Resume, Restart, Stop) specified in the `Decider`
 * to the child actor that failed, as opposed to [[akka.actor.AllForOneStrategy]] that applies
 * it to all children.
```